test/e2e/test_config.py

```diff
@@ -0,0 +1,92 @@
+import typing
+from subprocess import Popen
+import time
+import json
+from urllib.request import Request, urlopen
+import pytest
+from mitmproxy import http, ctx
+
+
+@pytest.fixture(scope="module")
+def mitm(tmpdir_factory):
+    tmpdir = tmpdir_factory.mktemp("mitm")
+    with open(tmpdir / "config.yaml", "w") as f:
+        f.write(_YAML_CONFIG)
+    mitmdump = Popen([
+        "mitmdump",
+        "--set", "confdir=" + str(tmpdir),
+        "--scripts", __file__,
+    ])
+    time.sleep(5)
```

Comment:
yeah, agree.
It might not be necessary, I haven't dived that deep. If needed it could be made resilient by having retries for a request or connection timeout might be enough (?).

Reply:
I think the better way here is to not send any requests at all. We can start the proxy with one script that just prints out the option values, and a second one that immediately terminates the proxy once it has been started (`mitmproxy/data/addonscripts/shutdown.py`). Or do you see any problems with that?

Reply:
I think it would be great if we can added e2e/integration tests for requests and responses too, not only config.